File: problems/826_most_profit_assigning_work/Solution.java
import java.util.*;

class Solution {
    public int maxProfitAssignment(int[] difficulty, int[] profit, int[] worker) {
        int n = difficulty.length;
        int[][] jobs = new int[n][2];
        for (int i = 0; i < n; i++) jobs[i] = new int[]{difficulty[i], profit[i]};
        Arrays.sort(jobs, (a, b) -> a[0] - b[0]);
        Arrays.sort(worker);
        int result = 0, best = 0, j = 0;
        for (int w : worker) {
            while (j < n && jobs[j][0] <= w) best = Math.max(best, jobs[j++][1]);
            result += best;
        }
        return result;
    }
}

File: problems/826_most_profit_assigning_work/solution.py
from typing import List


class Solution:
    def maxProfitAssignment(self, difficulty: List[int], profit: List[int], worker: List[int]) -> int:
        jobs = sorted(zip(difficulty, profit))
        worker.sort()
        result = 0
        best = 0
        j = 0
        for w in worker:
            while j < len(jobs) and jobs[j][0] <= w:
                best = max(best, jobs[j][1])
                j += 1
            result += best
        return result
